File: java/src/main/java/com/aerospike/helper/query/GenerationQualifier.java
package com.aerospike.helper.query;

import com.aerospike.client.Value;
/**
 * Qualifier used to query by generation
 * @author peter
 *
 */
public class GenerationQualifier extends Qualifier {

	public GenerationQualifier(FilterOperation op, Value value) {
		super(QueryEngine.Meta.GENERATION.toString(), op, value);
	}
	@Override
	protected String luaFieldString(String field) {
		return "generation";
	}

}

File: csharp/AerospikeHelper/AerospikeHelper/Query/GenerationQualifier.cs
using System;
using Aerospike.Client;

namespace Aerospike.Helper.Query
{	
	/// <summary>
	/// Qualifier used to query by generation
	/// </summary>
	public class GenerationQualifier : Qualifier
	{
		public GenerationQualifier(FilterOperation op, Value value): base(QueryEngine.Meta.GENERATION.ToString(), op, value){
		}

		protected override string luaFieldString(String field) {
			return "generation";
		}


	}
}

